app/models/user.rb

```diff
@@ -1,9 +1,12 @@
 class User < ApplicationRecord
   # Existing associations
+  has_many :libraries, dependent: :destroy
+  has_many :blogs, dependent: :destroy
+  has_many :gift_cards, dependent: :destroy
+  has_many :user_chanels, dependent: :destroy
   has_many :sender_messages,
            class_name: 'Message',
            foreign_key: :sender_id, dependent: :destroy
```

Comment:
The 'email' field validation allows blank values due to 'allow_blank: true', which contradicts the 'presence: true' validation. Remove 'allow_blank: true' to ensure the email cannot be blank.